node/src/transition_frontier/transition_frontier_state.rs

```diff
@@ -196,4 +196,19 @@ impl TransitionFrontierState {
             reorg_best_tip: false, // TODO: Unused for now
         })
     }
+
+    pub fn resources_usage(&self) -> serde_json::Value {
+        serde_json::json!({
+            "best_chain_size": self.best_chain.len(),
+            "needed_protocol_states_size": self
+                .needed_protocol_states
+                .len(),
+            "blacklist_size": self.blacklist.len(),
+            "diff_tx_size": self
+                .chain_diff
+                .as_ref()
+                .map(|d| d.new_commands.len() + d.removed_commands.len())
```

Comment:
same here, and I think it will be the case for all these mem usage calculations, saturating ops